### ISSUE TITLE
v2-run needs a way to use package directory

### ISSUE THREAD

**Issue by symbiont-sam-halliday** [haskell/cabal#6178]
**Describe the bug**

`v2-run tasty` uses the current working directory when invoking the `tasty` binary, but this breaks multi-package builds (i.e. projects with a `cabal.project`) that have test data in a specific location relative to the `.cabal` file.

For example, `aeson-golden` tests will fail because they expect to be run from the package directory, not the location of the `cabal.project` file.

**To Reproduce**

```
$ cd foo
$ cabal v2-run tasty
```

vs

```
$ cd ..
$ cabal v2-run foo:tasty
```

**Expected behavior**

The tests should run with the CWD set to the directory containing the `.cabal` file.

**System information**
 - Operating system: N/A
 - `cabal`, `ghc` versions: all released versions

** Other Context

Not only does this mean that running tests from the root dir vs the package dir produce different results, but `stack` uses the convention of always running tests from the dir containing the `.cabal` file and therefore this subtle change in behaviour makes it difficult to migrate from stack to cabal or for developers to have a choice of build tool on their project.

**Comment by phadej** [comment]
I’m pretty sure tests are run from pkg directory, otherwise `cabal v2-test Cabal:parser-tests` would fail. Which version of cabal-install do you have, is it a regression?

**Comment by symbiont-sam-halliday** [comment]
@phadej apologies, you're right... my reproduction is broken. This is actually a problem with using `cabal v2-run foo:tasty` to run tests, which is currently required because it is not possible to pass parameters such as `-p Pattern` to `v2-test`. I will update the description. (I wrote this bug report from some old notes, I got lazy and should have confirmed!)

An alternative solution to the core problem would be if something like `cabal v2-test -O0 --tasty-options=-p Pattern` worked as expected.
